### PR TITLE
Added Active_pull_request_commentors

### LINKED ISSUE #931: Script to fetch active Pull Request Commentors
# Description

Create a Script to fetch active Pull Request Commentors using Github's GraphQL API. The responses and analysis can be saved in a CSV file or a graph can be generated for quick visualization.

## Language

- [X] Go
- [X] Javascript
- [X] Python

### PR DESCRIPTION
**Pull request by sourabhkumar45**
JavaScript code to access data for PR commentors

# Description

Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.

Fixes #931 

## Type of change

_Please delete options that are not relevant._

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation Update

# Checklist:

- [ ] My code follows the style guidelines(Clean Code) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have created a helpful and easy to understand `README.md`
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by vybhav72954 — CHANGES_REQUESTED** [review]
Please properly link the issue in the body of PR.

**Comment by sourabhkumar45** [comment]
> Please properly link the issue in the body of PR.

Added the link to the issue.

File: JavaScript/Active_pull_request_commentors/output.csv [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+https://github.com/HarshCasper/Rotten-Scripts,937,https://github.com/HarshCasper/Rotten-Scripts/pull/937,"@vybhav72954
+I am really sorry, I didn't know about this.
+Should I make any changes currently for this PR?",Dhanush2612
+https://github.com/HarshCasper/Rotten-Scripts,937,https://github.com/HarshCasper/Rotten-Scripts/pull/937,"@vybhav72954
+I am really sorry, I didn't know about this.
+Should I make any changes currently for this PR?
+
+Nopes, nothing as of now, am yet to review it. But please ensure to properly link issue next time and try to use PR template, they are here to help you out only.",vybhav72954
+https://github.com/HarshCasper/Rotten-Scripts,937,https://github.com/HarshCasper/Rotten-Scripts/pull/937,"@vybhav72954
+Thank you",Dhanush2612
```
Comment (HarshCasper) [inline JavaScript/Active_pull_request_commentors/output.csv:164]: Remove this

File: JavaScript/Active_pull_request_commentors/script.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
+const fetch = require("node-fetch");
+const createCsvWriter = require("csv-writer").createObjectCsvWriter;
+
+let OWNER = `HarshCasper`; // replace with repository's owner username
+let REPO_NAME = `Rotten-Scripts`; // replace with repositort's name
```
Comment (HarshCasper) [inline JavaScript/Active_pull_request_commentors/script.js:5]: Prefer to pass it dynamically to the Script

File: JavaScript/Active_pull_request_commentors/script.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+let NO_OF_REQUEST = 10; // replace with no. of latest Active pull request data you want to access
+let obj;
+const accessToken = ""; // replace with your own github access token
```
Comment (HarshCasper) [inline JavaScript/Active_pull_request_commentors/script.js:8]: Keep a `.env` file where the access token can be kept. Push a `.env.example` in the PR to help the users know how to fill the access token

File: JavaScript/Active_pull_request_commentors/script.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
+const fetch = require("node-fetch");
+const createCsvWriter = require("csv-writer").createObjectCsvWriter;
+
+let OWNER = `HarshCasper`; // replace with repository's owner username
+let REPO_NAME = `Rotten-Scripts`; // replace with repositort's name
+let NO_OF_REQUEST = 10; // replace with no. of latest Active pull request data you want to access
```
Comment (HarshCasper) [inline JavaScript/Active_pull_request_commentors/script.js:6]: Prefer to pass it dynamically to the Script

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: JavaScript/Active_pull_request_commentors/script.js (hunk 3 above)
Comment (sourabhkumar45) [reply JavaScript/Active_pull_request_commentors/script.js:8]: Do I have to provide the steps to generate personal github access token in .env.example itself?

File: JavaScript/Active_pull_request_commentors/readme.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  - give any title and all the permissions
+  - Be careful with the token and remember to delete after usual
+
+## How to use the code
+
+---
+
+- navigate to the folder where script.s file is.
+- in terminal or command prompt type - node script.js OWNER_NAME REPO_NAME NO_OF_REQUEST
+- the order of variables should be same as above
```
Comment (HarshCasper) [inline JavaScript/Active_pull_request_commentors/readme.md:29]: Can you use proper backticks to differentiate text from commands?

File: JavaScript/Active_pull_request_commentors/readme.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
+# Node.js Script to get Active Pull request commentors
+
+---
+
+## About
+
+---
```
Comment (HarshCasper) [inline JavaScript/Active_pull_request_commentors/readme.md:7]: Don't have so many empty lines

File: JavaScript/Active_pull_request_commentors/readme.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+This script written in JavaScript extracts the data like repository's link , pull request Number, link to the pull request, comments on the pull request and username of the coresponding commentors. This script extract data for open PRs only and no. of PR to extract data is handled by user which is 10 by default.
```
Comment (HarshCasper) [inline JavaScript/Active_pull_request_commentors/readme.md:9]: Can you run it against a Grammarly lint?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: JavaScript/Active_pull_request_commentors/script.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+              bodyText
+            }
+          }
+        }
+      }
+    }
+  }
+  `;
+
+fetch("https://api.github.com/graphql", {
```
Comment (sohamsshah) [inline JavaScript/Active_pull_request_commentors/script.js:36]: Fetch isn't the best practice in JS.
Modern ES6+ practices specify to use `async await` functions to perform fetch calls.
You can abstract this piece out into an async function and `await` on the https://api.github.com/graphql link and then perform try catch on that.

[This](https://javascript.info/async-await) will help you a lot

**Review by sohamsshah — CHANGES_REQUESTED** [review]
The script looks well to me. Good work! Just make the suggested changes and it is good to merge!

File: JavaScript/Active_pull_request_commentors/script.js (hunk 3 above)
Comment (HarshCasper) [reply JavaScript/Active_pull_request_commentors/script.js]: It would be preferred,

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**